src/mokernels/independent.jl

```diff
@@ -23,18 +23,44 @@ struct IndependentMOKernel{Tkernel<:Kernel} <: MOKernel
     kernel::Tkernel
 end
 
+# kernel function should be symmetric
+# would really like (κ::IndependentMOKernel)((x, px)::Tuple{T,Int}, (y, py)::Tuple{T,Int}) where T, but seems to cause autodiff problems
 function (κ::IndependentMOKernel)((x, px)::Tuple{Any,Int}, (y, py)::Tuple{Any,Int})
-    if px == py
-        return κ.kernel(x, y)
-    else
-        return 0.0
-    end
+    return κ.kernel(x, y) * (px == py)
```

Comment:
that seems like a good resolution!

Reply:
Or maybe even neater (because it more explicitly shows what we're evaluating here):
```suggestion
    return κ.kernel(x, y) * I[px, py]
```
I benchmarked it - same speed.